projects/js-themes-toolkit/packages/liferay-theme-tasks/theme/__tests__/index.test.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * SPDX-FileCopyrightText: © 2017 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: MIT
+ */
+
+const {r2} = require('liferay-theme-tasks');
+
+describe('index.js', () => {
+	it('exposes the r2 API from the "main" package resource', () => {
+		expect(require('liferay-theme-tasks')).toBe(require('..'));
+
+		expect(Object.keys(r2).sort()).toEqual(['exec', 'swap', 'valueMap']);
```

Comment:
And then showing what's in it has the expected shape.

Reply:
This feels like an original take to me. I get that it shows it has the expected shape, but we hardly ever do this, do we? What's the reason we want do really do it here?

We also know this will fail if we ever add other things to `r2` which I guess is expected?

Nothing blocking here, just looking to understand when would we want to assert exported APIs like this and when not.

Reply:
In this case acceptance criteria were "I can use r2 API from theme tasks".

Now, before this PR, you could already do that hackily with:

```js
const r2 = require('liferay-theme-tasks/lib/r2/liferay-r2');
```

but that's fragile (path dependent) and breaks encapsulation. So given that I didn't actually add any new functionality in this PR, I just changed how it was exposed, I wanted to make assertions about _how_ it was exposed. So that leads me to these assertions:

- I can get at the API without reaching "inside" or making assumptions about package structure.
- The API I get is actually the one I think I'm getting (but without re-exercising all of its behavior, because that is already covered elsewhere; that's why we have the quick `swap()` test below, as a smoke test).

So that's why we "hardly ever do this", as you say — most likely, we "never do this — and probably won't ever have to do it again.

Reply:
> We also know this will fail if we ever add other things to `r2` which I guess is expected?

And yes, that's right. I promise if this test ever fails for that reason, I will purchase you a beverage of your choice up to a monetary value of 5 _Lifebucks_.